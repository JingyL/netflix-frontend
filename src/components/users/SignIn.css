.signin-card{
  height: 520px;
  width: 400px;
  background-color: rgba(48, 1, 1, 0);
  position: absolute;
  transform: translate(-50%,-50%);
  top: 50%;
  left: 50%;
  border-radius: 10px;
  backdrop-filter: blur(10px);
  border: 2px solid rgba(255,255,255,0.1);
  box-shadow: 0 0 40px rgba(8,7,16,0.6);
  padding: 50px 35px;
}

.signin-label-font{
  color:white;
  font-family: 'Libre Baskerville', serif;
}

label{
  display: block;
  margin-top: 30px;
  font-size: 16px;
  font-weight: 500;
  text-align:left;
}

input{
  display: block;
  height: 50px;
  width: 100%;
  background-color: rgba(255,255,255,0.07);
  border-radius: 3px;
  padding: 0 10px;
  margin-top: 8px;
  font-size: 14px;
  font-weight: 300;
}

.signin-page-btn{
  margin-top: 50px;
  width: 100%;
  background-color: #ffffff;
  color: #080710;
  padding: 15px 0;
  font-size: 18px;
  font-weight: 600;
  border-radius: 5px;
  cursor: pointer;
  margin-bottom: 30px;
}

.signup-btn{
  text-decoration: none;
  color:brown;
}

/* .btn-style{
  background-color: #7F5539;
  color: #c7b7a9;
  margin-left: 20px;
  font-size: 20px;
} */

.danger{
  color:red;
}

.nav-sign-in-page{
  margin-top: 12px;
  position: fixed;
  top: 0;
  display: flex;
  width: 98%;
  /* justify-content: space-between; */
  padding-left: 15px;
}