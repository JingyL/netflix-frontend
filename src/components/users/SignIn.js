import React, { useState } from "react";
import { Link, useHistory } from "react-router-dom";
import "./SignIn.css"

// Sign in form
function SignIn({ login }) {
  const [formData, setFormData] = useState({
    username: "",
    password: "",
  });
  const [formErrors, setFormErrors] = useState("");

  function handleChange(e) {
    e.persist();
    console.log(e.target.value);
    setFormData(f => ({ ...f, [e.target.name]: e.target.value }));
  }


  const history = useHistory();

  async function handleSubmit(e) {
    e.preventDefault();
    // get login token, if errors, give err msg, else link to /
    let response = await login(formData);
    if (response["success"]) {
      history.push("/");
    } else {
      setFormErrors(response.error);
    }

  }


  return (

    <div className="signin">
      <div className="nav-sign-in-page">
        <img className="nav-logo" src="https://upload.wikimedia.org/wikipedia/commons/0/08/Netflix_2015_logo.svg" alt="Netflix Logo"></img>
      </div>
      <div className="signin-card">
        <div>
          <form>
            <div className="signin-label-font">
              <label>Username</label>
              <input
                name="username"
                value={formData.username}
                onChange={handleChange}
                required
              />
            </div>
            <div className="signin-label-font">
              <label>Password</label>
              <input
                name="password"
                value={formData.password}
                onChange={handleChange}
                required
              />
            </div>

            {formErrors != ""
              ? <p className="danger" messages={formErrors}>{formErrors}</p>
              : null}

            <button className="signin-page-btn" onClick={handleSubmit}>
              Sign In
            </button>

            <Link to="/signup" className="signup-btn">
              Sign Up
            </Link>

          </form>
        </div>
      </div>
    </div>





  );
}

export default SignIn;