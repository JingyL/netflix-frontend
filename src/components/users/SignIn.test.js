import React from "react";
import { render } from "@testing-library/react";
import SignIn from "./SignIn";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <SignIn />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
