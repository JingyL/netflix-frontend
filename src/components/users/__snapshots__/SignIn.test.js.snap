// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="signin"
  >
    <div
      class="nav-sign-in-page"
    >
      <img
        alt="Netflix Logo"
        class="nav-logo"
        src="https://upload.wikimedia.org/wikipedia/commons/0/08/Netflix_2015_logo.svg"
      />
    </div>
    <div
      class="signin-card"
    >
      <div>
        <form>
          <div
            class="signin-label-font"
          >
            <label>
              Username
            </label>
            <input
              name="username"
              required=""
              value=""
            />
          </div>
          <div
            class="signin-label-font"
          >
            <label>
              Password
            </label>
            <input
              name="password"
              required=""
              value=""
            />
          </div>
          <button
            class="signin-page-btn"
          >
            Sign In
          </button>
          <a
            class="signup-btn"
            href="/signup"
          >
            Sign Up
          </a>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
